### DISCUSSION TITLE
Is it possible to set the agent's location for predfined tasks?

### DISCUSSION THREAD

**Discussion by Bpoole908** [PKU-Alignment/safety-gymnasium#112] [Q&A]
I was simply wondering if it is possible to set agent's `locations` attribute for the predefined tasks. I see there is a `locations` attribute, but the `Underlying` class builds the agent instance using the `_build_agent()` method. However, when building the agent, no class kwargs can be passed. This makes it seem like it is impossible to pass any agent class attributes.  Is there any way around this?

My use case is for fixing the agent's position and requiring it to start from a specific location, not a random location.

**Comment by muchvo** [comment]
I'm very sorry for overlooking this discussion. Yes, you can certainly fix the position of any object. Please refer to [this parameter](https://github.com/PKU-Alignment/safety-gymnasium/blob/main/safety_gymnasium/bases/base_agent.py#L162-L163). Here’s how you could use this feature:
```python
class GoalLevel0(BaseTask):
    """An agent must navigate to a goal."""

    def __init__(self, config) -> None:
        super().__init__(config=config)

        self.placements_conf.extents = [-1, -1, 1, 1]

        self._add_geoms(Goal(keepout=0.305))

        self.last_dist_goal = None
        self.agent.locations = [(0, 0)] # location
        self.agent.rot= np.pi/2 * 0 # rotation
```

**Reply by Bpoole908** [reply]
No worries, I actually ended up modifying this [pull request](https://github.com/PKU-Alignment/safety-gymnasium/pull/100) so that the agent can be configured via the passed config as well.

**Reply by muchvo** [reply]
Sounds good! I look forward to seeing what you come up with. 😊
